### PR TITLE
[RFC] Introduce empty_string helper function

### PR DESCRIPTION
**Pull request by bstoeger**
### Describe the pull request:

- [ ] Bug fix
- [ ] Functional change
- [ ] New feature
- [x] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:

I noticed that there are ca. 50 constructs of the kind `same_string(s, "")` to check for empty or NULL string, so I thought that maybe this warrants an individual helper `empty_string()` (commit 2). Of course, the old version is also quite expressive, so I'm a bit undecided with this commit.

Commit 3 then additionally deploys this helper to a dozen or so manual checks of the kind `s && *s` for consistency reason. If we don't want `empty_string()`, these should probably be replaced by `same_string(s, "")`.

Commit 1 and 4 are minor cleanups which are independent of the rest: Let string-comparison functions return bool and make an ominous loop-construct more clear.

Note: This compiles, but is only lightly tested.

### CODE REVIEW

File: core/file.c [hunk 1]
```diff
@@ -650,7 +650,8 @@ int parse_txt_file(const char *filename, const char *csv)
 		cur_cylinder_index++;
 
 		lineptr = strstr(memtxt.buffer, "Dive started at");
-		while (!empty_string(lineptr) && (lineptr = strchr(lineptr, '\n')) && ++lineptr) {
+		while (!empty_string(lineptr) && (lineptr = strchr(lineptr, '\n'))) {
+			++lineptr;	// Skip over '\n'
```
Comment (neolit123) [inline core/file.c]: seems like a semi-bug fix even?

**Review by neolit123 — APPROVED** [review]
thanks, lgtm.

File: core/file.c (hunk 1 above)
Comment (bstoeger) [reply core/file.c]: Hmmm... Now that you say it, perhaps this was originally intended as
```
while (!empty_string(lineptr) && (lineptr = strchr(lineptr, '\n')) && *++lineptr)
```
?
In any case, `*lineptr == '\0'` is handled gracefully in `next_mkvi_key()`, so not an issue...

Comment (neolit123) [reply core/file.c]: `*++lineptr` was probably was the intended.

**Comment by neolit123** [comment]
any further comments? @dirkhh @mturkia

**Review by dirkhh — APPROVED** [review]
This looks good. Why is it still marked as RFC?
Otherwise I would have merged it.

**Comment by mturkia** [comment]
LGTM

**Pull request merged**